### PR TITLE
feat: update pipeline for vegacapsule network

### PR DESCRIPTION
**Pull request by daniel1302**
# Passed pipeline

https://jenkins.ops.vega.xyz/job/private/job/Deployments/job/Vegacapsule/job/Stagnet%203/116/console

# Changes

- Adjusting pipeline for stagnet3. 
- Introducing a fix for a bug (https://github.com/vegaprotocol/vegacapsule/issues/228)
- Got rid of env variables in favor of call parameters for the capsule pipeline wrapper - testing with env variables was not very easy, and anyway, values are very stable. Moreover I feel like the env variables hidden in the job config generate confusion when you look on the pipeline code where they are coming from

### CODE REVIEW

File: vars/capsulePipelineWrapper.groovy [hunk 1]
```diff
@@ -1,245 +1,485 @@
...
+      branch: 'main',
+      directory: version,
+      credentialsId: ghCredentialsID
+  ])
+  sh label: 'Building ' + binType,
+        script: 'go mod vendor && go build -o ' + outDir + '/' + binType + ' ' + packagesMapping[binType]
+}
+
+void downloadS3Binary(String s3Path, String binName, String outDir) {
+  sh label: 'Download the ' + binName + ' from the S3',
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:76]: Please use interpolation instead of string concats

File: vars/capsulePipelineWrapper.groovy [hunk 2]
```diff
@@ -1,245 +1,485 @@
...
+        'vega': 'cmd/vega/',
+        'vegawallet': 'cmd/vegawallet/',
+        'data-node': 'cmd/data-node/'
+    ]
+
+  if (!packagesMapping.containsKey(binType)) {
+    error('[compileBinary] invalid binary type. got: "' + binType + '"')
+  }
+
+  gitClone([
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:65]: Nit: you don't need to add [ ] when calling function which takes only map as argument

File: vars/capsulePipelineWrapper.groovy [hunk 3]
```diff
@@ -1,245 +1,485 @@
...
+  sh label: 'Set permissions on the ' + binName + ' binary',
+       script: 'chmod a+x "' + outDir + '/' + binName + '"'
+}
+
+boolean isS3Link(String path) {
+    /* groovylint-disable-next-line DuplicateNumberLiteral */
+  return path.length() > 0 && path[0..4] == 's3://'
+}
+
+void veganetSh(String network, String command, String sshCredentialsId, String dockerCredentialsId) {
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:87]: Let's extract this method to external file and use it together with pipelineDeploy to reduce copy paste

File: vars/capsulePipelineWrapper.groovy [hunk 4]
```diff
@@ -1,245 +1,485 @@
...
+          sh label: 'Create markets',
+            script: './veganet.sh ' + network + ' ' + command
+        }
+      }
     }
+  }
+}
+
+void call(Map customConfig = [:]) {
+  Map config = [
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:108]: If there are no default values let's skip this definition and use config only taken as function argument

File: vars/capsulePipelineWrapper.groovy [hunk 5]
```diff
@@ -1,245 +1,485 @@
...
+    networkName: '',
+    nomadAddress: '',
+    awsRegion: '',
+    vegacapsuleS3BucketName: '',
+  ] + customConfig
+
+  pipeline {
+    agent any
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:116]: Maybe it's worth asking ourselves question if there's any specific agent label we want to use for capsule

File: vars/capsulePipelineWrapper.groovy [hunk 6]
```diff
@@ -1,245 +1,485 @@
...
+      NETWORK_SSH_CREDENTIALS = 'ubuntu-ansible-key'
+    }
 
-            AWS_SECRET_ACCESS_KEY = credentials('jenkins-vegacapsule-aws-secret')
-            AWS_ACCESS_KEY_ID = credentials('jenkins-vegacapsule-aws-id')
+    stages {
+      stage('Prepare') {
+        steps {
+          script {
+            print('Parameters: ' + params)
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:140]: Nit: if you do echo "parameters: ${params}" you don't need script block, as echo is step where print is groovy function. in most cases Jenkins prefers steps over functions

File: vars/capsulePipelineWrapper.groovy [hunk 7]
```diff
@@ -1,245 +1,485 @@
...
-                    expression {
-                        params.ACTION == 'START' || params.ACTION == 'RESTART'
-                    }
-                }
-                steps {
-                    sh "vegacapsule network start --home-path '${env.CONFIG_HOME}'"
-                }
+            steps {
+              withGHCLI('credentialsId': env.GITHUB_CREDS) {
+                sh 'gh release --repo vegaprotocol/vegacapsule download ' + params.VEGACAPSULE_VERSION + ' --pattern "*linux*"'
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:200]: You did a function for Vega binary using GH release command, let's parametrize it and call it here too

File: vars/capsulePipelineWrapper.groovy [hunk 8]
```diff
@@ -1,245 +1,485 @@
...
+            --home-path '""" + netHome + """' \
+            --path '""" + tmplHome + """/data_node.full.tmpl.toml' \
+            --nodeset-group-name full \
+            --type data-node \
+            --with-merge """ + additionalFlags
+}
+
+void getReleasedBinary(String ghCredentialsID, String binType, String version) {
+  withGHCLI('credentialsId': ghCredentialsID) {
+        sh 'gh release --repo vegaprotocol/vega download ' + version + ' --pattern "' + binType + '-linux*"'
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:46]: As you reported issue with GH command let's add retry here

File: vars/capsulePipelineWrapper.groovy [hunk 9]
```diff
@@ -1,245 +1,485 @@
...
+            }
+          }
+        }
+      }
+
+      stage('Sync remote state to local') {
+        steps {
+          dir('networks-internal/' + config.networkName + '/vegacapsule') {
+            sh '''
+                mkdir -p "./home"
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:302]: Interpolation ?

File: vars/capsulePipelineWrapper.groovy [hunk 10]
```diff
@@ -1,245 +1,485 @@
...
+            'bounce_bots',
+            env.NETWORK_SSH_CREDENTIALS,
+            env.GITHUB_CREDS,
+          )
+        }
+      }
+    }
+
+    post {
+      always {
```
Comment (mkjmdski) [inline vars/capsulePipelineWrapper.groovy:480]: Do we want to send slack message?

Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:519]: We do :) Let me add it

File: vars/capsulePipelineWrapper.groovy (hunk 8 above)
Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:46]: I have retry in the step where this function is used. I did not wanted to make it overcomplicated

File: vars/capsulePipelineWrapper.groovy (hunk 3 above)
Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:87]: fixing

File: vars/capsulePipelineWrapper.groovy (hunk 4 above)
Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:108]: fixed

File: vars/capsulePipelineWrapper.groovy (hunk 5 above)
Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:118]: no, there is no agent. We have everything on standard linux server :)

File: vars/capsulePipelineWrapper.groovy (hunk 7 above)
Comment (daniel1302) [reply vars/capsulePipelineWrapper.groovy:186]: fixed

**Review by daniel1302 — COMMENTED** [review]
Merging as We need. If something requires fix, let's create followup actions

**Pull request merged**